File: problem-solutions/SummerSchool/semana2/dia4/B.py
import sys

def fastpow(b, e):
    if(int(e) == 0):
        return 1

    x = fastpow(int(b), int(e)/2)
    if(int(e)%2 == 0): 
        return int(x)*int(x)
    else: 
        return int(b)*int(x)*int(x)

fat = [1 for i in range(402)]

def precalc():
    i = 1
    while(i < 402):
        fat[i] = i*fat[i-1]
        i = i+1

precalc()
n= int(input())

res = fat[n] * fat[2*n]/(fat[n]*fat[n])
res -= 1

i = 1
while(i < n):
    res -= fat[n]/(fat[n-i]*fat[i]) * fat[2*n - (n-i)]/(fat[n-(n-i)]) * fat[2*n - (n-i)]
    i = i+1

print(res)

File: problem-solutions/SummerSchool/semana2/dia4/B.java
import java.io.*;
import java.math.*;
import java.util.*;

class B{

	static BigInteger fastpow(BigInteger b, BigInteger e){
		if(e.equals(BigInteger.valueOf(0))) return BigInteger.valueOf(1);

		BigInteger x = fastpow(b, e.divide(BigInteger.valueOf(2)));
		if((e.mod(BigInteger.valueOf(2))).equals(BigInteger.valueOf(0))){
			return x.multiply(x);
		}
		else{
			return b.multiply(x.multiply(x));
		}
	}

	static void precalc(BigInteger fat[]){
		fat[0] = BigInteger.valueOf(1);

		for(int i = 1; i < 402; i++){
			fat[i] = fat[i-1].multiply(BigInteger.valueOf(i));
		}
	}

	public static void main(String[] args) throws Exception{
		BigInteger fat[] = new BigInteger[402];
		precalc(fat);

		Scanner input = new Scanner(System.in);
		BigInteger b = input.nextBigInteger();
		BigInteger e = input.nextBigInteger();

		BigInteger res = fat[n.intValue()].multiply(fat[2*n.intValue()].divide(fat[n.intValue()].multiply(fat[n.intValue()])));

		input.close();
	}
}
